### PR TITLE
serviceconnector: Add Service Connector UI

### PR DESCRIPTION
**Pull request by motm32**
This is how the items appear in the tree: 
image

### CODE REVIEW

File: serviceconnector/resources/02833-icon-menu-Resource-Connector.svg
Comment (alexweininger) [inline serviceconnector/resources/02833-icon-menu-Resource-Connector.svg:1]: Can you rename this file to something cleaner?

File: serviceconnector/src/tree/ServiceConnectorGroupTreeItem.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    public async loadMoreChildrenImpl(): Promise<AzExtTreeItem[]> {
+        const client = new (await import('@azure/arm-servicelinker')).ServiceLinkerManagementClient(this.subscription.credentials);
+        const linkers = (await uiUtils.listAllIterator(client.linker.list(nonNullValue(this.resourceId))));
+
+        const children = await this.createTreeItemsWithErrorHandling(
+            linkers,
+            'invalidServiceConnector',
+            l => {
+                return new ServiceConnectorTreeItem(nonNullValue(this.parent), l, this)
```
Comment (alexweininger) [inline serviceconnector/src/tree/ServiceConnectorGroupTreeItem.ts:33]: So technically the grandparent and the parent are being passed here to `ServiceConnectorTreeItem`. Shouldn't it only need `this` (its parent)?

File: serviceconnector/src/tree/ServiceConnectorItem.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    const id = `${item.id}/ServiceConnector/${linker.name}`;
+
+    return {
+        id,
+        subscription,
+        linker,
+        item,
+        viewProperties: {
+            data: linker,
+            label: `${item.id} ${vscode.l10n.t('linker', 'Linker')} ${linker.name}`,
```
Comment (alexweininger) [inline serviceconnector/src/tree/ServiceConnectorItem.ts:31]: the label doesn't need the id, can just be the linker name if you want

File: serviceconnector/src/tree/ServiceConnectorTreeItem.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    public readonly item: AzExtTreeItem;
+
+    constructor(parent: AzExtParentTreeItem, linker: LinkerResource, item: AzExtTreeItem) {
+        super(parent);
+        this.linker = linker;
+        this.item = item;
+    }
+
+    public get id(): string {
+        return `${this.item.id}/ServiceConnector/${this.linker.name}`;
```
Comment (alexweininger) [inline serviceconnector/src/tree/ServiceConnectorTreeItem.ts:23]: Might be good to create a shared method for getting the ids

**Review by alexweininger — APPROVED** [review]

File: serviceconnector/src/tree/ServiceConnectorGroupItem.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import { uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { ISubscriptionContext, callWithTelemetryAndErrorHandling, createSubscriptionContext, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { ResourceModelBase, ViewPropertiesModel } from "@microsoft/vscode-azureresources-api";
+import * as vscode from 'vscode';
+import { TreeItem, TreeItemCollapsibleState } from "vscode";
+import { LinkerItem } from "../createLinker/createLinker";
+import { getIconPath } from "./IconPath";
+import { ServiceConnectorItem, createServiceConnectorItem } from "./ServiceConnectorItem";
+
+export interface TreeElementBase extends ResourceModelBase {
```
[the remark this answers is not in the document]
Comment (alexweininger) [reply serviceconnector/src/tree/ServiceConnectorGroupItem.ts:15]: It's already in utils

[the remark this answers is not in the document]
Comment (nturinski) [reply serviceconnector/src/tree/ServiceConnectorGroupItem.ts:15]: Oh then... Megan, just use that.

**Review by nturinski — APPROVED** [review]

**Pull request merged**